### ISSUE TITLE
jax.scipy.stats.multinomial.logpmf/pmf return -inf for all rows of a batched input (sum reduces over all axes, not axis=-1)

### ISSUE THREAD

**Issue by OE-GOD** [jax-ml/jax#38605]
### Description

`jax.scipy.stats.multinomial.logpmf` (and `.pmf`) return `-inf` / `0` for **every row** of a batched input, even when each row is a valid count vector summing to `n`:

```python
import jax; jax.config.update('jax_enable_x64', True)
import numpy as np, jax.scipy.stats.multinomial as jmn
import scipy.stats as ss
p = np.array([0.2, 0.3, 0.5]); x = np.array([[2,3,5],[1,4,5]])
print(jmn.logpmf(x, 10, p))        # [-inf -inf]
print(ss.multinomial.logpmf(x,10,p))  # [-2.46451596 -2.75219803]
# per-row JAX is correct:
print(jmn.logpmf(np.array([2,3,5]),10,p), jmn.logpmf(np.array([1,4,5]),10,p))
# -2.4645159601402717 -2.7521980325920516
```

Each row sums to `n=10`, so the pmf is finite and positive; `-inf` is wrong.

### Root cause

`jax/_src/scipy/stats/multinomial.py::logpmf` ends with a validity mask:

```python
return jnp.where(jnp.equal(jnp.sum(x), n), logprobs, -np.inf)
```

`jnp.sum(x)` reduces over **all** axes (here 2+3+5+1+4+5 = 20), not `axis=-1` (which would give `[10, 10]`). Since `20 != 10`, the mask is `False` for every row and all outputs are forced to `-inf`. Per-row (unbatched) calls are correct because the full sum then equals `n`.

### Suggested fix

Reduce the count over the last axis: `jnp.sum(x, axis=-1)` in the mask, broadcasting against `n`.

### Cross-check

`scipy.stats.multinomial.logpmf` and `torch.distributions.Multinomial(10, probs=p).log_prob(x)` both return `[-2.4645, -2.7522]`; JAX is the lone outlier for batched inputs.

### System info
```
jax 0.10.1, jaxlib 0.10.1, numpy 2.4.6, python 3.12.13, CPU
```

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
